### PR TITLE
Custom Storybook theme for Recharts brand

### LINKED ISSUE #3446: Storybook: Recharts Style in Storybook.
The storybook should fit in better with the website. https://recharts.org/en-US/

Change it such that it better fits with the brand.
This issues does not have explicit acceptance criteria but might require multiple smaller changes. 

https://storybook.js.org/docs/react/configure/theming

### PR DESCRIPTION
**Pull request by csdiehl**
## Description

Creates a custom Storybook theme using the colors, fonts, and logo from Recharts.org. Links to Recharts.org if user clicks on the logo.  

Note: I grabbed a screenshot of the logo from the website. Might want to replace with a better image if one is available. 

## Related Issue

Closes #3446. 

## Motivation and Context

Customizes Storybook to better line up with the website branding. 

## How Has This Been Tested?

This only affects the Storybook theme, not the function of any chart components. All tests are passing. 

## Types of changes

- [ X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a storybook story or extended an existing story to show my changes
- [ X] All new and existing tests passed.

### CODE REVIEW

File: storybook/RechartsTheme.js
Comment (ckifer) [inline storybook/RechartsTheme.js:3]: Could we make this a TS file instead of a JS file?

File: storybook/RechartsTheme.js
Comment (ckifer) [inline storybook/RechartsTheme.js:2]: The 'logo' isn't really a logo, its just an H1 with some styling 

See here https://github.com/recharts/recharts.org/blob/master/src/layouts/Frame.js#L63-L67

Not sure if that works in storybook's case though

File: storybook/manager.js
Comment (ckifer) [inline storybook/manager.js:3]: this should also be a TS file

**Review by ckifer — COMMENTED** [review]
We should be able to use TS instead of JS everywhere.

Also not sure why but the build is failing

File: storybook/RechartsTheme.js
Comment (csdiehl) [reply storybook/RechartsTheme.js:2]: yeah I saw that - but Storybook requires an image. If you have one hosted somewhere I can link to that instead

File: package-lock.json
Comment (ckifer) [inline package-lock.json:4565]: @csdiehl This PR is using your companies internal NPM registry - can you make the package json changes and install with normal npm?

**Review by ckifer — CHANGES_REQUESTED** [review]
This should be why the build is failing

**Comment by csdiehl** [comment]
@ckifer  Everything should be ok now, except the brand image. Has to be a link to a hosted image, doesn't allow it to be an html element. When you have that, we can swap it in

**Comment by ckifer** [comment]
Yeah things look good besides the brand image, it doesn't seem to load for me at all there.

Maybe we just keep it the default for now? Recharts doesn't really have an official logo, but @wesoudshoorn worked on one for us and its in the contributor slack ([invite link](https://join.slack.com/t/recharts/shared_invite/zt-1o9t7luva-x9CbZ1_nI2OHXMr6id1T4A)) we can probably swap to that sooner or later (Wes if you have any input let us know)

I'm going to wait for some other folks to review this before merging but it looks good overall to me. Definitely will look better in the website - it does get rid of any 'dark mode', but thats probably okay cause the site doesn't have one anyways.
(screenshot for others so they don't have to pull changes down)
image

CC @nikolasrieble @PavelVanecek

File: storybook/RechartsTheme.ts
Comment (nikolasrieble) [inline storybook/RechartsTheme.ts:3]: Wonderful, this is great. Thank you! 
I am not reviewing each and every value here, simply because this PR is a big step in the right direction and sets up the configuration. 

We can independently iterate on the exact values to be used.

**Review by nikolasrieble — APPROVED** [review]
❤️ 🚀 🚢

**Pull request merged**
